spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/DefaultDbRefResolver.java

```diff
@@ -106,7 +107,7 @@ public DBRef createDbRef(org.springframework.data.mongodb.core.mapping.DBRef ann
 	 */
 	@Override
 	public DBObject fetch(DBRef dbRef) {
-		return ReflectiveDBRefResolver.fetch(mongoDbFactory.getDb(), dbRef);
+		return ReflectiveDBRefResolver.fetch(MongoClientVersion.isMongo3Driver() ? mongoDbFactory.getDb() : null, dbRef);
```

Comment:
Wouldn't the test case become much simple if we just handed the factory into the fetch method? `ReflectiveDbRefResolver` does an `isMongo3Driver()` anyway, so why not just lookup the `DB` within that guard? That would be a simple unit test for `ReflectiveDbRefResolver.fetch(MongoDbFactory, DBRef)` then, wouldn't i?